<hr>
<b><font color='red'>Ejercicio 01</font></b>  

Lee el archivo.txt 

In [51]:
archivo="archivo.txt"
text=open(archivo)

while True:
    linea = text.readline()
    if not linea: 
        break
    print("  "+linea[0]+".0  "+ linea[2]+".00")


    


  1.0  2.00
  2.0  4.00
  3.0  6.00
  4.0  8.00


y haz que tenga la siguiente salida:

    1.0   2.00  
    2.0   4.00  
    3.0   6.00  
    4.0   8.00  


<hr>
<b><font color='red'>Ejercicio 02</font></b>  

Haz lo mismo con el archvo archivo_error.txt,con ciddado de que a este archivo le falta algún dato, haz que en ese caso cuando lo lea, si no hay dos columnas no lo escriba.

In [207]:
archivo="archivo_error.txt"
text=open(archivo)

while True:
    linea = text.readline()
    if not linea: 
        break
    if(len(linea)>=4):
        print("  "+linea[0]+".0  "+ linea[2]+".00")

  1.0  2.00
  3.0  6.00
  4.0  8.00


<hr>
<b><font color='red'>Ejercicio 03</font></b>  

Lee el archivo.txt, haz que imprima las columnas en las filas y las filas en las columnas

In [114]:
archivo="archivo.txt"
text=open(archivo)
a=[]
c=""
d=""
while True:
    linea = text.readline()
    if not linea: 
        break
    for i in range(0,2):
        a.append(linea.split()[i])
        
for i in range(len(a)):
    if(i % 2 == 0):
        c += (a[i]+" ")
    else:
        d += (a[i] +" ")
        
print(c)
print(d)


1 2 3 4 
2 4 6 8 


<hr>
<b><font color='red'>Ejercicio 04</font></b>  

1. Escribe en un archivo la siguiente matriz:
$
\begin{equation}
\begin{pmatrix}
3 & 1 & 3\\
-1 & 1 & 0\\
-2 & 4 & 1
\end{pmatrix}
\end{equation} 
$
2. Lee el archivo
3. Calcula con los datos leidos el determinante de la matriz y lo saque por pantalla

In [267]:
f = open("matriz.txt","w")
f.write("3 1 3\n-1 1 0\n-2 2 1")
f.close()
#--------------------------
file= open("matriz.txt","r")
matriz=[]

for line in file:
    matriz += [line.split()]

def determinante(matriz):
    sum1=int(matriz[0][0])*int(matriz[1][1])*int(matriz[2][2])
    sum2=int(matriz[0][1])*int(matriz[1][2])*int(matriz[2][0])
    sum3=int(matriz[0][2])*int(matriz[1][0])*int(matriz[2][1])
    res1=int(matriz[0][2])*int(matriz[1][1])*int(matriz[2][0])
    res2=int(matriz[0][1])*int(matriz[1][0])*int(matriz[2][2])
    res3=int(matriz[0][0])*int(matriz[1][2])*int(matriz[2][1])
    resultadoFinal = sum1 + sum2 + sum3 - res1 - res2 - res3
    return resultadoFinal
    
print(determinante(matriz))


4


<hr>
<b><font color='red'>Ejercicio 05</font></b>  

En el criptoanálisis, la técnica de análisis de frecuencias consiste en el aprovechamiento de estudios sobre la frecuencia de las letras o grupos de letras en los idiomas para poder establecer hipótesis para aprovecharlas para poder descifrar un texto cifrado sin tener la clave de descifrado (romper). Es un método típico para romper cifrados clásicos. 
Haz una anailisis de frecuencias para los 10 primeros càpitulos de el libro "el_quijote.txt"

In [261]:
abc="ABCDEFGHIJKLMNÑOPQRSTUVWXYZ"
archivo=open("el_quijote.txt",encoding='utf8')
f={}
sumTodasLetras=0
for i in abc:
    f[i]=0

archivoFiltrado=archivo.read().upper().replace("Á","A").replace("É","E").replace("Í","I").replace("Ó","O").replace("Ú","U")

for line in archivoFiltrado:
    if(line in abc):
        f[line] += 1
        sumTodasLetras= sumTodasLetras +1


for i in abc:
    print("%.2c  " % i,"%.3f" % (f[i]/sumTodasLetras))

A   0.127
B   0.017
C   0.037
D   0.053
E   0.138
F   0.005
G   0.011
H   0.012
I   0.053
J   0.007
K   0.000
L   0.060
M   0.026
N   0.066
Ñ   0.000
O   0.099
P   0.021
Q   0.019
R   0.061
S   0.073
T   0.037
U   0.048
V   0.011
W   0.000
X   0.000
Y   0.015
Z   0.004


Salida:

    A   0.127
    B   0.017
    C   0.037
    D   0.053
    E   0.138
    F   0.005
    .
    .
    .

<hr>
<b><font color='red'>Ejercicio 06</font></b>  

haz un analisis de frecuencias en el libro encriptado "RH RQCEPESEFM AR MPM.txt", y comparala con el analisis anterior


In [264]:
abc="ABCDEFGHIJKLMNÑOPQRSTUVWXYZ"
archivo1=open("RH RQCEPESEFM AR MPM.txt", encoding="utf8")
archivo2=open("el_quijote.txt",encoding='utf8')
fRH={}
fQuijote={}
sumTodasLetrasRH=0
sumTodasLetrasQuijote=0

for i in abc:
    fRH[i]=0
    fQuijote[i]=0

archivoFiltrado1=archivo1.read().upper().replace("Á","A").replace("É","E").replace("Í","I").replace("Ó","O").replace("Ú","U")
archivoFiltrado2=archivo2.read().upper().replace("Á","A").replace("É","E").replace("Í","I").replace("Ó","O").replace("Ú","U")

for line in archivoFiltrado1:
    if(line in abc):
        fRH[line] += 1
        sumTodasLetrasRH= sumTodasLetrasRH +1

for line in archivoFiltrado2:
    if(line in abc):
        fQuijote[line] += 1
        sumTodasLetrasQuijote= sumTodasLetrasQuijote +1
        
print("el_quijote.txt","  RH RQCEPESEFM AR MPM.txt")    
for i in abc:
    print("%.2c  " % i,"%.3f" % (fQuijote[i]/sumTodasLetrasQuijote),"       %.2c  " % i,"%.3f" % (fRH[i]/sumTodasLetrasRH))

el_quijote.txt   RH RQCEPESEFM AR MPM.txt
A   0.127        A   0.049
B   0.017        B   0.006
C   0.037        C   0.040
D   0.053        D   0.061
E   0.138        E   0.126
F   0.005        F   0.007
G   0.011        G   0.000
H   0.012        H   0.052
I   0.053        I   0.032
J   0.007        J   0.012
K   0.000        K   0.068
L   0.060        L   0.001
M   0.026        M   0.093
N   0.066        N   0.027
Ñ   0.000        Ñ   0.011
O   0.099        O   0.010
P   0.021        P   0.066
Q   0.019        Q   0.069
R   0.061        R   0.136
S   0.073        S   0.017
T   0.037        T   0.044
U   0.048        U   0.045
V   0.011        V   0.009
W   0.000        W   0.000
X   0.000        X   0.002
Y   0.015        Y   0.009
Z   0.004        Z   0.004


salida:

    el_quijote.txt   RH RQCEPESEFM AR MPM.txt
    A   0.127        A   0.049
    B   0.017        B   0.006
    C   0.037        C   0.040
    .                .
    .                .
    .                .

<hr>
<b><font color='red'>Ejercicio 07</font></b>  

Comparando el analisis de frecuencia sabrías decir cual es el titulo del libro.  
RH RQCEPESEFM AR MPM

RESULTADO= EL ESCARABAJO DE ORO